### DISCUSSION TITLE
Better catalog number handling

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5637] [Feature Requests]
Catalog number formatting is really cumbersome currently. The system should be smart enough to reserve numbers and skip existing ones. Pre-formatted or string catalog numbers that include a number suffix should be automatically interpreted. Maybe it should show the previous catalog number somehow.

This request is just a general call for smarter catalog numbering, reservations, and the ability to skip existing.

**Comment by maxpatiiuk** [comment]
Hmm, I though there already was an issue for this. Can't find it now

**Comment by grantfitzsimmons** [comment]
> Another possible scenario is that they want the catalognumber of the object plus an auto-incrementing suffix. The field formatter doesn't allow for other fields to be included as far as I can tell?

This request serves to enhance _any_ formated field

**Requested By:** [Corinna P at CSIRO](https://app.asana.com/0/1201972579963460/1203584420764663/f)

**Comment by tlammer** [comment]
this has been a pretty constant request over the years

**Comment by grantfitzsimmons** [comment]
Continues to be a requested feature, especially looking forward to geological collections

**Comment by grantfitzsimmons** [comment]
@realVinayak wants to get started on this project soon

**Comment by bronwyncombs** [comment]
Paul Blom @ College of Idaho had a request to remove the leading zeros from the catalog number field on the form. They don't utilize auto-numbering and opted to remove the field format altogether to solve this. 

> For us the leading zeros make it more difficult to read the actual value so removing them would be helpful.  It might also be helpful, for display purposes only, to have commas show thousands breaks.
